diff -Naur enigma2-PLi/lib/python/Components/UsageConfig.py enigma2-PersianPrince/lib/python/Components/UsageConfig.py
--- enigma2-PLi/lib/python/Components/UsageConfig.py	2018-11-13 16:46:05.706207636 +0330
+++ enigma2-PersianPrince/lib/python/Components/UsageConfig.py	2018-11-13 16:46:54.639810255 +0330
@@ -215,7 +215,7 @@
 		choicelist.append((str(i), ngettext("%d minute", "%d minutes", m) % m))
 	config.usage.timeshift_start_delay = ConfigSelection(default = "0", choices = choicelist)
 
-	config.usage.alternatives_priority = ConfigSelection(default = "0", choices = [
+	config.usage.alternatives_priority = ConfigSelection(default = "5", choices = [
 		("0", "DVB-S/-C/-T"),
 		("1", "DVB-S/-T/-C"),
 		("2", "DVB-C/-S/-T"),
