diff --git a/servicemp3/servicemp3.cpp b/servicemp3/servicemp3.cpp
index 93e570e..0072eab 100644
--- a/servicemp3/servicemp3.cpp
+++ b/servicemp3/servicemp3.cpp
@@ -623,6 +623,12 @@ eServiceMP3::eServiceMP3(eServiceReference ref):
 #endif
 	if ( m_gst_playbin )
 	{
+	/*
+	*test forcing aml-video-sink
+	*/
+		videoSink = gst_element_factory_make("amlvsink", "GstDVBVideoSink");
+		g_object_set (m_gst_playbin, "video-sink", videoSink, NULL);
+		eDebug("setting video-sink=amlvsink");
 		/*
 		 * avoid video conversion, let the dvbmediasink handle that using native video flag
 		 * volume control is done by hardware, do not use soft volume flag
