diff -Naur servicemp3-PLi/servicemp3/servicemp3.cpp servicemp3-PersianPrince/servicemp3/servicemp3.cpp
--- servicemp3-PLi/servicemp3/servicemp3.cpp	2018-12-30 10:11:43.959849196 +0330
+++ servicemp3-PersianPrince/servicemp3/servicemp3.cpp	2018-12-30 10:12:43.718669863 +0330
@@ -623,6 +623,12 @@
 #endif
 	if ( m_gst_playbin )
 	{
+	/*
+	*test forcing aml-video-sink
+	*/
+		videoSink = gst_element_factory_make("amlvsink", "GstDVBVideoSink");
+		g_object_set (m_gst_playbin, "video-sink", videoSink, NULL);
+		eDebug("setting video-sink=amlvsink");
 		/*
 		 * avoid video conversion, let the dvbmediasink handle that using native video flag
 		 * volume control is done by hardware, do not use soft volume flag
